test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterBaseTests.cs

```diff
@@ -0,0 +1,242 @@
+                jsonWriter.WriteValue((short)10);
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeInt""");
+                jsonWriter.WriteValue(10);
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeLong""");
+                jsonWriter.WriteValue(10L);
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeDouble""");
+                jsonWriter.WriteValue(10.2);
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeFloat""");
+                jsonWriter.WriteValue(10.2f);
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeDecimal""");
+                jsonWriter.WriteValue(10.2m);
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeGuid""");
+                jsonWriter.WriteValue(new Guid("00000012-0000-0000-0000-012345678900"));
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeObject""");
+                jsonWriter.StartObjectScope();
+                jsonWriter.EndObjectScope();
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeDateTimeOffset""");
+                jsonWriter.WriteValue(new DateTimeOffset(2014, 12, 31, 12, 42, 30, new TimeSpan(1, 20, 0)));
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeDate""");
+                jsonWriter.WriteValue(new Date(2014, 12, 31));
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeTimeOfDay""");
+                jsonWriter.WriteValue(new TimeOfDay(12, 42, 30,100));
+                jsonWriter.EndArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.StartArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeTimeSpan""");
+                jsonWriter.WriteValue(new TimeSpan(12, 42, 30));
+                jsonWriter.EndArrayScope();
+                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeObject""");
+                jsonWriter.StartObjectScope();
+                jsonWriter.EndObjectScope();
+                jsonWriter.EndArrayScope();

```

Comment:
Did you really intend this:
```
[[\"rawAtArrayStartBeforeTimeSpan\",\"PT12H42M30S\"],\"rawAtArrayStartBeforeObject\",{}]
```
or this:
```
[\"rawAtArrayStartBeforeTimeSpan\",\"PT12H42M30S\"],[\"rawAtArrayStartBeforeObject\",{}]
```
If you intended the latter, _and my guts tells me that's what you intended_, I'd expect the code here to look like:
```suggestion
                jsonWriter.StartArrayScope();
                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeTimeSpan""");
                jsonWriter.WriteValue(new TimeSpan(12, 42, 30));
                jsonWriter.EndArrayScope();
                jsonWriter.StartArrayScope();
                jsonWriter.WriteRawValue(@"""rawAtArrayStartBeforeObject""");
                jsonWriter.StartObjectScope();
                jsonWriter.EndObjectScope();
                jsonWriter.EndArrayScope();
```